src/client/client.ts

```diff
@@ -3854,4 +3859,259 @@ export class Client {
+      )
+      .join('\n')
+    const mutationStr = `mutation placeLimitOrders($affiliateDeveloperCode: AffiliateDeveloperCode, ${paramNames}) {\n${aliases}\n}`
+    const mutation = gqlstring(mutationStr)
+
+    const variables = {
+      affiliateDeveloperCode: this.affiliateDeveloperCode
+    }
+
+    placeLimitOrderPayloads.forEach((item, index) => {
+      variables[`p${names[index]}`] = item.signedPayload
+      variables[`s${names[index]}`] = item.signature
+    })
+
+    try {
+      const result = await this.gql.mutate({
+        mutation,
+        variables
+      })
+
+      return {
+        orders: Object.keys(result.data).map(
+          k => result.data[k]
+        ) as OrderPlaced[]
+      }
+    } catch (e) {
+      return {
+        error: e,
+        orders: []
+      }
+    }
+  }
+
+  private generatePlaceOrdersParams = async (
+    params: PlaceLimitOrderParams[]
+  ): Promise<PayloadSignature[]> => {
+    return await Promise.all(
+      params.map(async param => {
+        const { nonceOrder, noncesFrom, noncesTo } = this.getNoncesForTrade(
+          param.marketName,
+          param.buyOrSell
+        )
+        const normalizedAmount = normalizeAmountForMarket(
+          param.amount,
+          this.marketData[param.marketName]
+        )
+        const normalizedLimitPrice = normalizePriceForMarket(
+          param.limitPrice,
+          this.marketData[param.marketName]
+        )
+        const placeLimitOrderParams = createPlaceLimitOrderParams(
+          param.allowTaker,
+          normalizedAmount,
+          param.buyOrSell,
+          param.cancellationPolicy,
+          normalizedLimitPrice,
+          param.marketName,
+          noncesFrom,
+          noncesTo,
+          nonceOrder,
+          param.cancelAt
+        )
+        const signedPayload = await this.signPayload(placeLimitOrderParams)
+        return signedPayload
+      })
+    )
+  }
+
+  /**
+   * Cancel a list of orders by ID.
+   *
+   * @param orderID[]
+   * @returns
+   *
+   * Example
+   * ```
+   * const cancelledOrders = await nash.cancelOrder(['11','12'])
+   * console.log(cancelledOrders)
+   * ```
+   */
+  public async cancelAndPlaceOrders(
+    orderIDs: string[],
+    marketName: string,
+    orders: PlaceLimitOrderParams[]
+  ): Promise<OrdersCancelledAndPlaced> {
+    const [a, b] = marketName.split('_')
+    await this.prefillRPoolIfNeededForAssets(
+      a as CryptoCurrency,
+      b as CryptoCurrency
+    )
+
+    const cancelOrdersPayloads = await Promise.all(
+      orderIDs.map(async id => {
+        const cancelOrderParams = createCancelOrderParams(id, marketName)
+        const signedPayload = await this.signPayload(cancelOrderParams)
+        return signedPayload
+      })
+    )
+
+    const placeLimitOrderPayloads = await this.generatePlaceOrdersParams(orders)
+
+    const allNames = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K']
```

Comment:
I think you could replace this with a name generator function of sorts. `generateAliases(n) => ['A', 'B', ...,]`